### PR TITLE
Replace span labels with StackDriver versions

### LINKED ISSUE #442: Stackdriver exporter HTTP method not populated in Stackdriver Trace UI
When using the Stackdriver exporter - the HTTP Method in the Stackdriver Trace list is not being populated.

stackdriver_issue

As you can see the top part of the trace list HTTP Method is empty - even though the http.method property is filled.

I'm probably using it wrong?

The version I'm using is: `0.2.0-alpha.148`

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
Resolves #442.

StackDriver uses some special labels to categorise traces, eg `/http/method`, which do not match our internal labels, eg `http.method`.

This PR replaces four attributes, if found, when generating the StackDriver trace data during `ExportAsync`.

The labels it replaces are:
OTel label => StackDriver label
`http.method` => `/http/method`
`http.host` => `/http/host`
`http.status_code` => `/http/status_code`
`http.user_agent` => `/agent`

(GitHub Markdown for Table didn't work 👎 )

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
CC: @simonz130

File: src/OpenTelemetry.Exporter.Stackdriver/Implementation/SpanExtensions.cs [hunk 1]
```diff
@@ -23,6 +25,14 @@ namespace OpenTelemetry.Exporter.Stackdriver.Implementation
 {
     internal static class SpanExtensions
     {
+        private static Dictionary<string, string> httpLabelsToRepalce = new Dictionary<string, string>
```
Comment (nmbro) [inline src/OpenTelemetry.Exporter.Stackdriver/Implementation/SpanExtensions.cs]:
```suggestion
        private static Dictionary<string, string> httpLabelsToReplace = new Dictionary<string, string>
```

File: src/OpenTelemetry.Exporter.Stackdriver/Implementation/SpanExtensions.cs [hunk 2]
```diff
@@ -76,6 +86,17 @@ public static Google.Cloud.Trace.V2.Span ToSpan(this SpanData spanData, string p
                 };
             }
 
+            // StackDriver uses different labels that are used to categorize spans
+            // replace attribute keys with StackDriver version
+            foreach (var entry in httpLabelsToRepalce)
```
Comment (nmbro) [inline src/OpenTelemetry.Exporter.Stackdriver/Implementation/SpanExtensions.cs]:
```suggestion
            foreach (var entry in httpLabelsToReplace)
```

**Comment by MikeGoldsmith** [comment]
@SergeyKanzhelev these are before and after screenshots of GCP Trace console.

Before:
image

After:
image

**Comment by nmbro** [comment]
@MikeGoldsmith looks like there's a `/http/url` too - that could be used for `http.path` maybe? https://cloud.google.com/trace/docs/viewing-details

**Comment by MikeGoldsmith** [comment]
Sure, I'll take a look at adding that one too.

**Comment by MikeGoldsmith** [comment]
@nmbro added `/http/url` :+1:

**Comment by MikeGoldsmith** [comment]
@SergeyKanzhelev this is ready for merging - thanks

**Comment by nmbro** [comment]
@MikeGoldsmith btw - found a whole list of references for predefined labels here: https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#TraceSpan

**Comment by nmbro** [comment]
and by the looks of it /agent is wrong it should be /http/user_agent doh XD

**Comment by nmbro** [comment]
I will create another issue for fixing that

**Comment by MikeGoldsmith** [comment]
Thanks @nmbro - my mistake on assuming `/agent` was for user agent. I'll fix that up in #470.

**Pull request merged**